dlt/destinations/bigquery/bigquery.py

```diff
@@ -240,7 +242,7 @@ def _create_load_job(self, table_name: str, write_disposition: TWriteDisposition
                 max_bad_records=0)
             with open(file_path, "rb") as f:
                 return self.sql_client.native_connection.load_table_from_file(
-                        f,
+                        io.BytesIO(gzip.compress(f.read())),  # network latency is the bottleneck so compress the data here
```

Comment:
can't believe that they do not compress in transit! there are a few things that you'd need to improve in order to merge it:
1, should be optional. two reasons (https://cloud.google.com/bigquery/docs/loading-data-cloud-storage-json#limitations)
- gzip file max size is 4GB
- what you do above reads the file fully into memory
2. it compresses no matter what is the file type. we are adding parquet and avro so I doubt it is going to work

to fix that:
1. check if file type is eligible: get file format from ParsedLoadJobFileName.parse(file_name) and then DataWriter.data_format_from_file_format().is_binary_format must be false
2. add a config flag to BigQueryClientConfiguration
3. open a compressed gzip stream on the file, do not read full file into mem
4. if you have (3) working you may enable the compress option by default